core/src/main/scala/com/pingcap/tispark/write/TiDBWriter.scala

```diff
@@ -32,28 +32,26 @@ object TiDBWriter {
     options.checkWriteRequired()
     TiExtensions.getTiContext(sparkSession) match {
       case Some(tiContext) =>
-        val conn = TiDBUtils.createConnectionFactory(options.url)()
-
-        try {
-          val tableExists = TiDBUtils.tableExists(conn, options)
-          if (tableExists) {
-            saveMode match {
-              case SaveMode.Append =>
-                TiBatchWrite.write(df, tiContext, options)
-
-              case _ =>
-                throw new TiBatchWriteException(
-                  s"SaveMode: $saveMode is not supported. TiSpark only support SaveMode.Append.")
-            }
-          } else {
-            throw new TiBatchWriteException(
-              s"table `${options.database}`.`${options.table}` does not exists!")
-            // TiDBUtils.createTable(conn, df, options, tiContext)
-            // TiDBUtils.saveTable(tiContext, df, Some(df.schema), options)
+        val catalog = tiContext.tiSession.getCatalog
+        val tableExists =
+          if (catalog.getTable(options.database, options.table) == null) false else true
```

Comment:
`val tableExists =  catalog.getTable(options.database, options.table) != null ` will be simper

Reply:
Thanks, I have changed it to `val catalog = tiContext.tiSession.getCatalog.getTable(options.database, options.table) != null`.